projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+    return fn(this.state);
+  }
+
+  dispatch(action) {
+    this.state = compute(this.state, action);
+    // add a way to broadcast this store change to listeners
+  }
+
+  compute(state, action) {
+    return {
+      language: languageReducer(state.language, action),
+      products: productsReducer(state.products, action) 
+    }
+  }
+}
+</code-example>
+We have on purpose left out the part on how we subscribe to the store and how we broadcast a new store state after it has been changed by a call to `dispatch()`. The reason for that is that it's implementation specific to what Redux implementation we are talking about. In some implementations EventEmitters are being used, in NGRX it's RxJS. 
+
+At this point we have mentioned all major concepts in Redux like `action`, `reducer` and `store`. We've even shown a very naive implementation so you understand how all the pieces go together. Next is understanding where NGRX comes with its specific solution and especially why the heavy usage of Typescript makes sense and helps creating a better experience.
+
+## Introducing NGRX and the need for type safety
+NGRX is a Redux implementation meant for Angular and it makes heave use of Typescript and RxJS. The biggest reason is probably typesafety. 
+
+So why would we want to make them typesafe in the first place? Well, remember when we constructed an action in our previous section and assigned a string value to the `type` field? That is actually quite error prone, cause you might mispel that string value. It's maybe hard to understand why a mispel would be such a big deal but let's very briefly mention how actions are used. 
+
+### Dispatching an action
+Actions are being *dispatched* to reducer functions. What does that mean? Normally when you set up NGRX you do so be setting up your store like this:
+
+<code-example header="app.module.ts and products.reducer.ts">
+// products.reducer.ts
+export function productsReducer(state = [], action) {
+  switch(action.type) {
+    case '[Products] add':
+      return [...state, action.payload]
+    default:
+      return state;
+  }
+}
+
+// app.module.ts
+StoreModule.forRoot({
+  products: productsReducer
+})
+</code-example>
+
+When you do that you say that `products` is a thing that will exist in a store and it's state is something we can either *read* or *change*. The way we *read* the value is by subscribing to it using the `Store` service inside of component usually like so:
+
+<code-example header="app.module.ts">
+constructor(private store: Store<AppState>) {
+  this.store.pipe(
+    select('products')
+  ).subscribe(products => console.log('products', products));
+}
+</code-example>
+
+The way we *change* we change our data in the store is by calling the same `Store` service and its `dispatch()` method. The `dispatch()` method will *dispatch*, i.e send an action to the `store` and try to match that to a reducer and specifically a matching case in a reducer. So if we in our component where to do this:
```

Comment:
```suggestion
The way we *change* our data in the store is by calling the same `Store` service and its `dispatch()` method. The `dispatch()` method will *dispatch*, i.e send an action to the `store` and try to match that to a reducer and specifically a matching case in a reducer. So if we in our component where to do this:
```

This reads like NgRx invokes the "correct" reducer, but it actually sends the action to all reducers. Just saying it here, because sometimes people have a confusion about this.

Reply:
good point, I'll make that more clear..